timer.js

```diff
@@ -25,48 +20,170 @@ function startTimer(durationSeconds = 25 * 60) {
     minutes = minutes < 10 ? "0" + minutes : minutes;
     seconds = seconds < 10 ? "0" + seconds : seconds;
 
-    // TODO store the minutes/seconds in state
-
     // updating the UI with the state
     document.getElementById("timerMinutes").textContent = minutes;
     document.getElementById("timerSeconds").textContent = seconds;
     document.getElementById("timer").classList.add("timer-active");
-
-    // decrement the timer by 1s
-    timerDurationS -= 1;
     // save the current remaining time to state
     state.timer.remainingTimeOnResume = timerDurationS;
-
+    console.log(state.timer.remainingTimeOnResume);
+    timerDurationS -= 1;
+    // decrement the timer by 1s
+    currentTomatoStartEnd();
     // handle when the timer runs out
     if (timerDurationS < 0) {
       clearInterval(state.timerIntervalId);
       if (state.isWorkInterval) {
         startTimer(300); // Start rest interval
         state.isWorkInterval = false;
+        updateCurrentTomato();
       } else {
-        updateTomatoCount();
         startTimer(1500); // Start work interval
         state.isWorkInterval = true;
       }
     }
   }, 1 * 1000);
-}
+  } else if (!state.isWorkInterval) {
+
+    let minutes = 0;
+    let seconds = 0;
+    
+    // activate the timer CSS
+    const timerDisplay = document.getElementById("timer");
+    timerDisplay.classList.add("timer-active"); // Add class when timer starts
+    
+    //   start the timer...
+    state.timerIntervalId = setInterval(function updateTimerCountdown() {
```

Comment:
I notice we have `updateTimerCountdown()` here and above on line 21 as well, are these functions similar enough that we could extract and reuse the same function? e.g.

```
function updateTimerCountdown() { ...handle updating timer here... }

state.timerIntervalId = setInterval(updateTimerCountdown, 1*1000)
```

or if we need to pass args into the function

```
function updateTimerCountdown(arg1) { ...handle updating timer here(arg1)... }

// anonymous function notation
state.timerIntervalId = setInterval(function(){
  updateTimerCountdown(arg1)
}, 1*1000)

// or with arrow function notation
state.timerIntervalId = setInterval(() => {
  updateTimerCountdown(arg1)
}, 1*1000)

// or with arrow function notation + implicit return (shortest notation)
state.timerIntervalId = setInterval(() => updateTimerCountdown(arg1), 1*1000)
```

Reply:
I am trying, but in this case countdown in not changing and it is looped in first timedurationS